BLT/BLT/WaterView.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  WaterView.swift
+//  BLT
+//
+//  Created by LorentzenN on 12/3/19.
+//  Copyright © 2019 Lorentzen. All rights reserved.
+//
+
+import UIKit
+import Foundation
+
+class WaterView: UIView, RenderTimerDelegate {
+    /// Path for view
+    var path: UIBezierPath!
+    /// Class to manage rendering the waves on a set interval
+    var renderTimer: RenderTimer = RenderTimer()
+    /// Start Position of Waves
+    var currentWaveStart: CGFloat = 0
+    /// Amplitude of Base Wave
+    var amplitude: CGFloat = 10
+    /// Number of Wave Layers
+    var numberOfWaves: Int = 3
+    /// Vertical Separation Between Waves
+    var waveVerticalSeparation: CGFloat = 20
+    /// Percent of Screen The Wave Will Take Up
+    var percentOfFrameFilled: Double = 30
+    /// Actual Starting Y Value for Base Wave
+    var baseYVal: CGFloat {
+        return self.frame.height - (CGFloat(self.percentOfFrameFilled) * 0.01 * self.frame.height)
+    }
+    //Wavelength of Wave in Pixels
+    var waveLength: CGFloat = -1
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+
+        self.backgroundColor = UIColor.groupTableViewBackground
```

Comment:
@jiahuac This is theoretically where the thread error triggers, but I'm not seeing an optional.